relay-server/src/services/project.rs

```diff
@@ -1137,102 +1138,79 @@ impl Project {
     pub fn check_buckets(
         &mut self,
         outcome_aggregator: Addr<TrackOutcome>,
-        buckets: Vec<Bucket>,
+        mut buckets: Vec<Bucket>,
     ) -> Option<(Scoping, ProjectMetrics)> {
-        match self.check_buckets_inner(outcome_aggregator, buckets) {
-            CheckedBuckets::Ok(scoping, metrics) => return Some((scoping, metrics)),
-            CheckedBuckets::ProjectExpired(len) => {
-                relay_log::error!(
-                    tags.project_key = self.project_key.as_str(),
-                    "there is no project state: dropping {len} buckets"
-                );
-            }
-            CheckedBuckets::ProjectDisabled(len) => {
-                relay_log::debug!("dropping {len} buckets for disabled project");
-            }
-
-            CheckedBuckets::NoScoping(len) => {
-                relay_log::error!(
-                    tags.project_key = self.project_key.as_str(),
-                    "there is no scoping: dropping {len} buckets"
-                );
-            }
-            CheckedBuckets::RateLimited(len) => {
-                relay_log::debug!("dropping {len} buckets due to rate limit");
-            }
-        };
-
-        None
-    }
-
-    /// Unit-testable helper function for [`Self::check_buckets`].
-    ///
-    /// Returns [`CheckedBuckets::Ok`] if metrics are currently allowed, or the reject reason otherwise.
-    fn check_buckets_inner(
-        &mut self,
-        outcome_aggregator: Addr<TrackOutcome>,
-        buckets: Vec<Bucket>,
-    ) -> CheckedBuckets {
-        let len = buckets.len();
-
         let Some(project_state) = self.valid_state() else {
-            return CheckedBuckets::ProjectExpired(len);
+            relay_log::error!(
+                tags.project_key = self.project_key.as_str(),
+                "there is no project state: dropping {} buckets",
+                buckets.len(),
+            );
+            return None;
```

Comment:
We have to call `reject_metrics` here as well, right? I know we didn't before either, but at least something we need to fix in the future.

I'll put that onto the metric stats todo

Reply:
Without a valid project state, I think we do not have sufficient information to build the scoping required to emit outcomes. Though we have at least partial scoping, so we could refactor this function to first obtain partial scoping and then use it until the full one is available.